pkg/auth/authproviders/registry_httphandler.go

```diff
@@ -227,6 +227,9 @@ func (r *registryImpl) tokenRefreshEndpoint(req *http.Request) (interface{}, err
 		httputil.SetCookie(httputil.ResponseHeaderFromContext(req.Context()), newRefreshCookie)
 	}
 
+	// Set the access token cookie for now.
+	httputil.SetCookie(httputil.ResponseHeaderFromContext(req.Context()), AccessTokenCookie(token))
```

Comment:
When refresh fails, at least for the reason that the underlying IdP refuses to refresh the session, we should reset the access token cookie.